ipatests/test_integration/test_ipahealthcheck.py

```diff
@@ -904,3 +910,201 @@ def test_ipahealthcheck_trust_agent_member_check(self):
+        """
+        instance = realm_to_serverid(self.master.domain.realm)
+        self.master.run_command(
+            [
+                "certutil",
+                "-L",
+                "-d",
+                paths.ETC_DIRSRV_SLAPD_INSTANCE_TEMPLATE % instance,
+                "-n",
+                "Server-Cert",
+                "-a",
+                "-o",
+                paths.ETC_DIRSRV_SLAPD_INSTANCE_TEMPLATE % instance
+                + "/Server-Cert.pem",
+            ]
+        )
+        self.master.run_command(
+            [
+                "certutil",
+                "-d",
+                paths.ETC_DIRSRV_SLAPD_INSTANCE_TEMPLATE % instance,
+                "-D",
+                "-n",
+                "Server-Cert",
+            ]
+        )
+        yield
+        self.master.run_command(
+            [
+                "certutil",
+                "-d",
+                paths.ETC_DIRSRV_SLAPD_INSTANCE_TEMPLATE % instance,
+                "-A",
+                "-i",
+                paths.ETC_DIRSRV_SLAPD_INSTANCE_TEMPLATE % instance
+                + "/Server-Cert.pem",
+                "-t",
+                "u,u,u",
+                "-f",
+                paths.IPA_NSSDB_PWDFILE_TXT,
+                "-n",
+                "Server-Cert",
+            ]
+        )
+
+    def test_ipahealthcheck_ipansschainvalidation(self, remove_server_cert):
+        """
+        Test for IPANSSChainValidation
+        """
+        error_msg = (
+            ': certutil: could not find certificate named "Server-Cert": '
+            "PR_FILE_NOT_FOUND_ERROR: File not found\n"
+        )
+        returncode, data = run_healthcheck(
+            self.master, "ipahealthcheck.ipa.certs", "IPANSSChainValidation",
+        )
+        assert returncode == 1
+        for check in data:
+            if check["kw"]["nickname"] == "Server-Cert":
+                assert check["result"] == "ERROR"
+                assert check["kw"]["reason"] == error_msg
+
+    @pytest.fixture
+    def remove_raagent_cert(self):
+        """
+        Fixture to rename RA agent cert and revert the change.
+        """
+        self.master.run_command(
+            ["mv", paths.RA_AGENT_PEM, paths.RA_AGENT_PEM + ".old"]
+        )
+        yield
+        self.master.run_command(
+            ["mv", paths.RA_AGENT_PEM + ".old", paths.RA_AGENT_PEM]
+        )
+
+    def test_ipahealthcheck_iparaagent(self, remove_raagent_cert):
+        """
+       Test for IPARAAgent
+       """
+        returncode, data = run_healthcheck(
+            self.master, "ipahealthcheck.ipa.certs", "IPARAAgent"
+        )
+        assert returncode == 1
+        for check in data:
+            assert check["result"] == "ERROR"
+            assert (
+                check["kw"]["msg"]
+                == "Unable to load RA cert: [Errno 2] No such file or "
+                "directory: '{}'".format(paths.RA_AGENT_PEM)
+            )
```

Comment:
This is fine but the RA agent checker does quite a bit more. It compares the cert in the LDAP people entry to the cert on disk and verifies that description matches the expectation: serial number, issue, subject.